src/main/java/com/unascribed/fabrication/mixin/f_balance/player_free_spawners/MixinMobSpawnerLogic.java

```diff
@@ -0,0 +1,22 @@
+package com.unascribed.fabrication.mixin.f_balance.player_free_spawners;
+
+import com.unascribed.fabrication.support.EligibleIf;
+import com.unascribed.fabrication.support.MixinConfigPlugin;
+import net.minecraft.world.MobSpawnerLogic;
+import org.spongepowered.asm.mixin.Mixin;
+import org.spongepowered.asm.mixin.injection.At;
+import org.spongepowered.asm.mixin.injection.Inject;
+import org.spongepowered.asm.mixin.injection.callback.CallbackInfoReturnable;
+
+@Mixin(MobSpawnerLogic.class)
+@EligibleIf(configEnabled="*.player_free_spawners")
+public class MixinMobSpawnerLogic {
+	
+	@Inject(method="isPlayerInRange()Z", at=@At(value="HEAD", target="Lnet/minecraft/world/MobSpawnerLogic;update()V"), cancellable=true)
+	public void isPlayerInRange(CallbackInfoReturnable<Boolean> cir) {
+		if (MixinConfigPlugin.isEnabled("*.player_free_spawners")) {
+			cir.setReturnValue(true);
+			cir.cancel();
```

Comment:
Er, one more note. Calling `setReturnValue` internally calls `cancel`, so this line is a no-op.